worlds/witness/player_items.py

```diff
@@ -148,39 +149,119 @@ def get_filler_items(self, quantity: int) -> Dict[str, int]:
+            good_doors = [
+                "Desert Doors & Elevator", "Keep Hedge Maze Doors", "Keep Pressure Plates Doors",
+                "Shadows Lower Doors", "Tunnels Doors",
+
+                "Keep Tower Shortcut (Door)", "Shadows Timed Door", "Tunnels Town Shortcut (Door)",
+
+                "Desert Panels", "Keep Hedge Maze Panels",
+
+                "Shadows Door Timer (Panel)", "Keep Hedge Maze 1 (Panel)", "Town Maze Stairs (Panel)",
+            ]
+
+            # While Caves Shortcuts don't unlock anything in symbol shuffle, you'd still rather have them early.
+            # But, we need to do some special handling for them.
+
+            if self._world.options.shuffle_doors in ("doors", "mixed"):
+                # Caves Shortcuts might exist in vanilla/panel doors because of "early caves: add to pool".
+                # But if the player wanted them early, they would have chosen "early caves: starting inventory".
+                # This is why we make sure these are "natural" Caves Shortcuts.
+                good_doors.append("Caves Shortcuts")
+
+            # These two doors are logically equivalent, so we choose a random one as to not give them twice the chance.
+            good_doors.append(
+                self._world.random.choice(["Caves Mountain Shortcut (Door)", "Caves Swamp Shortcut (Door)"])
+            )
+
+            if self._world.options.shuffle_vault_boxes:
+                good_doors.append("Windmill & Theater Doors")
+                if not self._world.options.shuffle_symbols:
+                    good_doors += [
+                        "Windmill & Theater Panels",
+
+                        "Windmill & Theater Control Panels",
+                    ]
+
+            if self._world.options.shuffle_doors == "doors":  # It's not as good in mixed doors because of Light Control
+                good_doors.append("Desert Light Room Entry (Door)")
+
+            if not self._world.options.shuffle_symbols:
+                good_doors += [
+                    "Bunker Doors", "Swamp Doors", "Glass Factory Doors", "Town Doors",
+
+                    "Bunker Entry (Door)", "Glass Factory Entry (Door)", "Symmetry Island Lower (Door)",
+
+                    "Bunker Panels", "Swamp Panels", "Quarry Outside Panels", "Glass Factory Panels"
+
+                    "Glass Factory Entry (Panel)",
+                ]
+
+                if self._world.options.shuffle_vault_boxes:
+                    good_doors += [
+                        "Windmill & Theater Panels",
+
+                        "Windmill & Theater Control Panels",
+                    ]
+
+            existing_doors = [door for door in good_doors if door in existing_items_lookup]
+
+            # On some options combinations with doors, there just aren't a lot of doors that unlock much early.
+            # In this case, we add some doors that aren't great, but are at least guaranteed to unlock 1 location.
+
+            fallback_doors = [
```

Comment:
Could also include: "Town Maze Stairs (Panel)", "Keep Shadows Shortcut (Door)", "Keep Shortcuts", "Keep Pressure Plates Doors", "Keep Tower Shortcut (Door)", "Tunnels Doors", "Tunnels Town Shortcut (Door)", "Quarry Stoneworks Doors", "Quarry Stoneworks Roof Exit (Door)", "Desert Doors & Elevator", "Desert Panels", "Keep Hedge Maze 1 Exit (Door)", "Keep Hedge Maze 2 Shortcut (Door)", "Keep Hedge Maze 3 Shortcut (Door)", "Keep Hedge Maze 4 Shortcut (Door)", "Keep Hedge Maze Doors", "Keep Hedge Maze 1 (Panel)", "Keep Hedge Maze Panel", "Shadows Laser Room Doors", "Shadows Timed Door", "Shadows Lower Doors", and "Shadows Door Timer (Panel)"

Reply:
I've considered all of these individually and added them accordingly. Let me know if you have any more suggested additions / changes